Extensions/FileSystem/filesystemtools.js

```diff
@@ -124,6 +124,87 @@ gdjs.fileSystem.makeDirectory = function (directory) {
   }
 }
 
+/**
+ * Save a string into a file.
+ * @param {string} text The string to be saved
+ * @param {string} savePath The absolute path on the filesystem
+ */
+gdjs.fileSystem.saveStringToFile = function (text, savePath) {
+  const fileSystem = typeof require !== 'undefined' ? require('fs') : null;
+
+  if (fileSystem) {
+    fileSystem.writeFile(savePath, text, (err) => {
+      if (err) {
+        console.error("Unable to save the text to path: '" + savePath + "': ", err);
+      }
+    });
+  }
+}
+
+/**
+ * Load a string from a file into a scene variable.
+ * @param {scenevar} scenevar The scene variable to store the string
+ * @param {string} loadPath The absolute path on the filesystem
+ */
+gdjs.fileSystem.loadStringFromFileSync = function (scenevar, loadPath) {
+  const fileSystem = typeof require !== 'undefined' ? require('fs') : null;
+
+  if (fileSystem) {
+    if (fileSystem.existsSync(loadPath)) {
```

Comment:
You don't need to check for file existence before reading it. It's actually discouraged by Node.js as a subtle race condition could happen (the file could get erased by another process AFTER the existsSync and BEFORE the readFile/readFileSync).

Instead, directly call fileSystem.readFileSync. In case of issue, an exception will be thrown and the `catch` branch will be reached - which is what you would expect (otherwise, you have a failure with a silent error).

Another point: I wonder if it's a good idea to store the error in an optional variable? I did this for Facebook Instant Games apis: https://github.com/4ian/GDevelop/blob/master/Extensions/FacebookInstantGames/facebookinstantgamestools.js#L131

The idea is that you have an 2nd variable, which is totally optional, that will be filled with the error code.
Can be useful for the game developers to display an error message if they find that the error variable was filled.